### PR TITLE
Infer fucose

### PR DESCRIPTION
**Pull request by armaan-abraham**

### CODE REVIEW

File: maserol/relative.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import numpy as np
```
Comment (cyrillustan) [inline maserol/relative.py:1]: Give this file a more specific name (something like `subclass_amount.py`?)

File: maserol/figures/figure_infer_fucose.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    axes, fig = getSetup((9, 4.5), (1, 2))
+    gp120 = load_file("data-glycan-gp120")
+    abs = HIgGFs
+    cube_alter = prepare_data(alter()["Fc"], data_id="alter")
+    gp120_ag = [ag for ag in cube_alter.Antigen.values if "gp120" in ag]
+    abundance_list = plot_ab_aggs(cube_alter.sel(Antigen=gp120_ag, Sample=gp120["subject"].values), abs, 3, 650, metric="mean_rcp", ax=axes[0])
+
+    measured_fucose_ratio = gp120["F.total"]
+    fucose_abs = [ab for ab in abs if ab.endswith("f")]
+    get_aggs = lambda abs: np.sum(np.sum(np.mean(np.array([abund.sel(Antibody=abs) for abund in abundance_list]), axis=0), axis=2), axis=1)
```
Comment (cyrillustan) [inline maserol/figures/figure_infer_fucose.py:21]: Would be helpful to add a comment on how your `inferred_fucose_ratio` is defined

File: maserol/preprocess.py [hunk 3]
```diff
@@ -29,14 +31,35 @@ def prepare_data(data: xr.DataArray, remove_rcp=None):
...
+    wanted_receptors = data.Receptor.values
+    if data_id is None:
+        # Receptors: only keep those with "IgGx" or "FcgRx"
+        wanted_receptors = [x for x in wanted_receptors if re.match("^igg", x, flags=re.IGNORECASE)] + \
+                           [x for x in wanted_receptors if re.match("fc[gr]*", x, flags=re.IGNORECASE) and x != "FcRalpha"]
+        if remove_rcp != None:
+            for r in remove_rcp:
+                wanted_receptors.remove(r)
+        data = data.sel(Receptor=wanted_receptors)
+    else:
```
Comment (cyrillustan) [inline maserol/preprocess.py:43]: You probably have a good reason for setting this up, but it's also helpful to keep things simple

**Review by cyrillustan — APPROVED** [review]
Looks good. Left a few comments. You can merge after addressing them

File: maserol/preprocess.py (hunk 3 above)
Comment (armaan-abraham) [reply maserol/preprocess.py:43]: The alternative is specifying this information each time a dataset is loaded, so I think the added complexity is worth it (this logic is pretty generic, so I'm guessing the yaml file will be added to in the future).

**Pull request merged**